Formatting fails with NoSuchElementException

## Configuration ##

```
version = 3.5.9
runner.dialect = scala213
fileOverride {
  "glob:**/src-2.12/**" {
    runner.dialect = scala212
  }
}

style = default
maxColumn = 80

project.git = true
project.excludeFilters = []

assumeStandardLibraryStripMargin = true

align {
  preset = some
  stripMargin = true
  openParenCallSite = false
  openParenDefnSite = false
  openParenTupleSite = true
  closeParenSite = true
  beforeOpenParenCallSite = true
  beforeOpenParenDefnSite = true
  arrowEnumeratorGenerator = true
  tokens = [ ":", "=", "=>", "->", "<-", "//" ]
}

danglingParentheses {
  defnSite = false
  callSite = true
  ctrlSite = true
  tupleSite = true
}

newlines {
  beforeOpenParenDefnSite = fold
  sometimesBeforeColonInMethodReturnType = true
  alwaysBeforeElseAfterCurlyIf = true
  inInterpolation = avoid
  beforeTypeBounds = unfold
  beforeCurlyLambdaParams = multilineWithCaseOnly
  afterCurlyLambdaParams = squash
  implicitParamListModifierPrefer = before
  avoidForSimpleOverflow=[tooLong,punct,slc]
}

includeNoParensInSelectChains = true

optIn {
  breakChainOnFirstMethodDot = true
  breaksInsideChains = true
}

importSelectors = singleLine
```

## Command-line parameters ##

```
scalafmt --debug shims/test/src-2.13/peschke/shims/resource/ManagedTest.scala
```

## Steps
Due to the nature of the issue, I'm unsure what code is causing the trouble. I have used a regex to check for trailing `,` before `)` or braces, and haven't found any issues there. The code also compiles and runs successfully, so whatever is causing the problem is at least valid Scala 2 code.

## Problem

`scalafmt` fails with a `NoSuchElementException`:

```
Using core runner [3.5.9]
Working directory: .../scala-commons
parsed config (v3.5.9)
Files to be formatted:
...scala-commons/shims/test/src-2.13/peschke/shims/resource/ManagedTest.scala
org.scalafmt.cli.FailedToFormat: .../scala-commons/shims/test/src-2.13/peschke/shims/resource/ManagedTest.scala
Caused by: java.util.NoSuchElementException: last of empty list
	at scala.collection.immutable.Nil$.last(List.scala:665)
	at scala.collection.immutable.Nil$.last(List.scala:661)
	at org.scalafmt.internal.FormatTokens.tokenAfter(FormatTokens.scala:199)
	at org.scalafmt.internal.Router.$anonfun$getSplitsImpl$94(Router.scala:759)
	at scala.Option.map(Option.scala:242)
	at org.scalafmt.internal.Router.getSplitsBeforeOpenParen$1(Router.scala:759)
	at org.scalafmt.internal.Router.$anonfun$getSplitsImpl$102(Router.scala:782)
	at scala.Option.map(Option.scala:242)
	at org.scalafmt.internal.Router.getSplitsImpl(Router.scala:772)
	at org.scalafmt.internal.Router.getSplits(Router.scala:2597)
	at org.scalafmt.internal.BestFirstSearch.$anonfun$routes$1(BestFirstSearch.scala:37)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:563)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:561)
	at org.scalafmt.internal.FormatTokens.foreach(FormatTokens.scala:15)
	at org.scalafmt.internal.BestFirstSearch.<init>(BestFirstSearch.scala:37)
	at org.scalafmt.internal.BestFirstSearch$.apply(BestFirstSearch.scala:344)
	at org.scalafmt.Scalafmt$.$anonfun$doFormatOne$6(Scalafmt.scala:156)
	at scala.util.Try$.apply(Try.scala:210)
	at org.scalafmt.Scalafmt$.$anonfun$doFormatOne$4(Scalafmt.scala:156)
	at scala.meta.parsers.Parsed.fold(Errors.scala:12)
	at scala.meta.parsers.Parsed.fold$(Errors.scala:11)
	at scala.meta.parsers.Parsed$Success.fold(Errors.scala:26)
	at org.scalafmt.Scalafmt$.org$scalafmt$Scalafmt$$doFormatOne(Scalafmt.scala:152)
	at org.scalafmt.Scalafmt$.doFormat(Scalafmt.scala:126)
	at org.scalafmt.Scalafmt$.formatCodeWithStyle(Scalafmt.scala:91)
	at org.scalafmt.Scalafmt$.$anonfun$formatCode$3(Scalafmt.scala:78)
	at scala.util.Success.fold(Try.scala:281)
	at org.scalafmt.Scalafmt$.formatCode(Scalafmt.scala:78)
	at org.scalafmt.cli.ScalafmtCoreRunner$.unsafeHandleFile(ScalafmtCoreRunner.scala:69)
	at org.scalafmt.cli.ScalafmtCoreRunner$.handleFile(ScalafmtCoreRunner.scala:53)
	at org.scalafmt.cli.ScalafmtCoreRunner$.$anonfun$run$5(ScalafmtCoreRunner.scala:36)
	at org.scalafmt.cli.ScalafmtCoreRunner$.$anonfun$run$5$adapted(ScalafmtCoreRunner.scala:35)
	at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:563)
	at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:561)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1293)
	at scala.collection.parallel.ParIterableLike$Foreach.leaf(ParIterableLike.scala:938)
	at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:52)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:97)
	at scala.collection.parallel.Task.tryLeaf(Tasks.scala:55)
	at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:49)
	at scala.collection.parallel.ParIterableLike$Foreach.tryLeaf(ParIterableLike.scala:935)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$AWSTWrappedTask.compute(Tasks.scala:152)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$AWSTWrappedTask.compute$(Tasks.scala:148)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$AWSFJTWrappedTask.compute(Tasks.scala:304)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```


## Expectation

Some indication of the line and/or column that it failed on would be very helpful when figuring out what code is causing scalafmt trouble.

By bisecting the code I was able to find the offending bit of syntax, which turned out to be a case class without parameters.

This minimized file reproduces the issue:
```scala
package example

sealed abstract class InvariantADT[A]
object Record {
  final case class IsOk[A](a: A) extends InvariantADT[A]
  final case class AlsoOk[A](msg: String) extends InvariantADT[A]
  final case class NotOk[A]() extends Record[A]
}
```

Removing the empty `()` produces an error message which instructs the user to add them back, so it seems like this is a case the formatter should be able to handle gracefully:
```
.../scala-commons/Temp.scala:7: error: [dialect scala213] case classes must have a parameter list; try 'case class NotOk()' or 'case object NotOk'
  final case class NotOk[A] extends Record[A]
```

--- a/scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala
+++ b/scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala
@@ -756,7 +756,7 @@ class Router(formatOps: FormatOps) {
               )
           }
           val argsOpt = if (isAlignFirstParen) getArgsOpt else None
-          argsOpt.map(x => tokens.tokenAfter(x).right).fold(splits) { x =>
+          argsOpt.flatMap(tokens.tokenAfterOpt).map(_.right).fold(splits) { x =>
             val noSplitIndents = Seq(
               Indent(StateColumn, x, ExpiresOn.Before),
               Indent(-indentLen, x, ExpiresOn.Before)
@@ -781,7 +781,7 @@ class Router(formatOps: FormatOps) {
               }
               getSplitsBeforeOpenParen(x, indent, _.beforeOpenParenDefnSite) {
                 rightOwner match {
-                  case SplitDefnIntoParts(_, _, _, args) => Some(args.last)
+                  case SplitDefnIntoParts(_, _, _, args) => args.lastOption
                   case _ => None
                 }
               }

--- a/scalafmt-tests/src/test/resources/default/Class.stat
+++ b/scalafmt-tests/src/test/resources/default/Class.stat
@@ -551,3 +551,15 @@ object a {
   }
 
 }
+<<< #3304
+align.closeParenSite = true
+align.beforeOpenParenDefnSite = true
+newlines.beforeOpenParenDefnSite = fold
+===
+object Record {
+  final case class IsOk[A](a: A) extends InvariantADT[A]
+  final case class AlsoOk[A](msg: String) extends InvariantADT[A]
+  final case class NotOk[A]() extends Record[A]
+}
+>>>
+org.scalafmt.util.FormatException: java.util.NoSuchElementException: last of empty list

--- a/scalafmt-tests/src/test/resources/default/Class.stat
+++ b/scalafmt-tests/src/test/resources/default/Class.stat
@@ -562,4 +562,8 @@ object Record {
   final case class NotOk[A]() extends Record[A]
 }
 >>>
-org.scalafmt.util.FormatException: java.util.NoSuchElementException: last of empty list
+object Record {
+  final case class IsOk[A](a: A) extends InvariantADT[A]
+  final case class AlsoOk[A](msg: String) extends InvariantADT[A]
+  final case class NotOk[A]() extends Record[A]
+}

